iis-wf/iis-wf-referenceextraction/src/main/java/eu/dnetlib/iis/wf/referenceextraction/softwareurl/HttpServiceFacade.java

```diff
@@ -55,89 +54,85 @@
-            case HttpURLConnection.HTTP_SEE_OTHER: {
-                String redirectedUrl = getHeaderValue(httpResponse.getAllHeaders(), HEADER_LOCATION);
-                if (StringUtils.isNotBlank(redirectedUrl)) {
-                    log.info("got {} response code, redirecting to {}, server response: {}", statusCode,
-                            redirectedUrl, EntityUtils.toString(httpResponse.getEntity()));
-                    return retrieveUrlContent(redirectedUrl, ++retryCount);
-                } else {
-                    return new ContentRetrieverResponse(
-                            new RuntimeException("resource was moved, missing redirect header for the url: " + currentUrl));
+                case HttpURLConnection.HTTP_OK: {
+                    HttpEntity entity = httpResponse.getEntity();
+                    if (!isResponseEntityValid(entity)) {
+                        return FacadeContentRetrieverResponse.persistentFailure(new HttpServiceFacadeException(
+                                "got empty entity in response, server response: " + httpResponse));
+                    }
+                    return readPageContent(entity, maxPageContentLength, url);
+                }
+                case HttpURLConnection.HTTP_MOVED_PERM:
+                case HttpURLConnection.HTTP_MOVED_TEMP:
+                case HttpURLConnection.HTTP_SEE_OTHER: {
+                    String redirectedUrl = getHeaderValue(httpResponse.getAllHeaders(), HEADER_LOCATION);
+                    if (StringUtils.isNotBlank(redirectedUrl)) {
+                        log.info("got {} response code, redirecting to {}, server response: {}", statusCode,
+                                redirectedUrl, httpResponse);
+                        return retrieveContentOrThrow(redirectedUrl, ++retryCount);
+                    }
+                    return FacadeContentRetrieverResponse.persistentFailure(new HttpServiceFacadeException(
+                            "resource was moved, missing redirect header for the url: " + url));
+                }
+                case HttpURLConnection.HTTP_NOT_FOUND: {
+                    return FacadeContentRetrieverResponse.persistentFailure(new HttpServiceFacadeException(
+                            "unable to find page at: " + url));
+                }
+                case HTTP_TOO_MANY_REQUESTS: {
+                    log.warn("got {} response code, potential reason: rate limit reached. Delaying for {} ms, server response: {}",
+                            statusCode, throttleSleepTime, httpResponse);
+                    Thread.sleep(throttleSleepTime);
+                    return retrieveContentOrThrow(url, ++retryCount);
+                }
+                default: {
+                    return FacadeContentRetrieverResponse.transientFailure(new HttpServiceFacadeException(String.format(
+                            "got unsupported HTTP response code: %d when accessing page at url: %s", statusCode, url)));
```

Comment:
The comment related to patent retriever and `default` section is applicable here as well. Let's discuss whether we should use `persistentFailure()` or `transientFailure()` here.